functions/src/utils/auth.js

```diff
@@ -0,0 +1,35 @@
+const oauth2 = require("simple-oauth2");
+
+/**
+ * Create a configured simple-oauth2 client.
+ *
+ * @see https://github.com/lelylan/simple-oauth2/blob/3.x/API.md#createoptions--module
+ */
+const createOAuth2Client = async ({
+  clientId,
+  clientSecret,
+  tokenHost,
+  tokenPath,
+  revokePath,
+  authorizeHost,
+  authorizePath,
+}) => {
+  const credentials = {
+    client: {
+      id: clientId,
+      secret: clientSecret,
+    },
+    auth: {
+      tokenHost,
+      tokenPath,
+      revokePath,
+      // TODO: how can we ensure we add the correct redirect URL here to revokePath? Maybe use a placeholder in the revokePath string or similar?
```

Comment:
[TODO] how can we ensure we add the correct redirect URL here to revokePath? Maybe use a placeholder in the revokePath string or similar?

This looks like:

```
/oauth/logout?redirect_uri=TODO_full_url_here
```

Probably return to the platform at `/v/${venueId}`?

Reply:
Did this.. not sure of the commit hash off the top of my head.